client/src/app/shared/video/video.service.ts

```diff
@@ -212,6 +215,12 @@ export class VideoService implements VideosProvider {
     if (categoryOneOf) params = params.set('categoryOneOf', categoryOneOf + '')
     if (skipCount) params = params.set('skipCount', skipCount + '')
 
+    if (nsfw) {
```

Comment:
If `nsfw` is a `NSFWPolicy`, you could do something like:

```
const nsfwPolicy = nsfwParam || peertubeLocalStorage.getItem(User.KEYS.NSFW_POLICY)
if (nsfwPolicy === 'do_not_list') ...
else if (nsfwPolicy === 'display' || nsfwPolicy === 'blur') ...
```

If the NSFW policy is set, you must set the parameter, or the server will fallback to the instance default setting